### DISCUSSION TITLE
Is `out of range assuming EOF` an error I should report to the language server?

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#11119] [Q&A]
In bash with this code:

```sh
#!/bin/bash

# Check if an argument is provided
if [ $# -eq 0 ]; then
  echo "Please provide an argument of 'o'verwrite or 'a'ppend."
  exit 1
fi

# Switch statement
case $1 in
o)
  sgpt --code --no-cache "{1}"
  ;;
a)
  echo "You entered 2."
  ;;
*)
  echo "Invalid argument. Please provide 1 or 2."
  ;;
esac

cat <<"_end_of_text"
Your user name is: $(whoami)
Your current working directory is: $PWD
our Bash version is: $BASH_VERSION
_end_of_text

col_green=$'\e[32;01m'
```
I get `2024-07-09T20:23:21.828 helix_lsp::util [WARN] LSP position Position { line: 2147483647, character: 2147483647 } out of range assuming EOF` in the log.

![image](https://github.com/helix-editor/helix/assets/12832280/ac32d698-ac4a-44dc-99c3-8d14890d12c0)

**Comment by the-mikedavis** [comment]
It looks like it's setting the line+char to the equivalent of `i32::MAX` and relying on the client to treat that position as the EOF. We do treat it as EOF so this is benign. We could probably change that log down to an `info` or maybe even `debug` level. Maybe @pascalkuthe has more insight on this, I believe he was looking at last in https://github.com/helix-editor/helix/pull/6512

A little bit off-topic: `i32::MAX` is a little curious because the spec notes that those fields are unsigned ints, so I would expect `u32::MAX`. But maybe they send `i32::MAX` to help clients that don't adhere to the spec closely.
